examples/quickstart_aws/quickstart_aws/assets/hackernews.py

```diff
@@ -38,40 +54,65 @@ def hackernews_topstories(
 
     df = pd.DataFrame(results)
 
-    # Dagster supports attaching arbitrary metadata to asset materializations. This metadata will be
-    # shown in the run logs and also be displayed on the "Activity" tab of the "Asset Details" page in the UI.
-    # This metadata would be useful for monitoring and maintaining the asset as you iterate.
-    # Read more about in asset metadata in https://docs.dagster.io/concepts/assets/software-defined-assets#recording-materialization-metadata
-    context.add_output_metadata(
-        {
+    # write the dataframe to an S3 bucket
+    s3.get_client().put_object(
+        Body=df.to_csv(index=False),
```

Comment:
I think we should create the df before trying to write the object, just in case the df.to_csv fails, it'll be easier to debug.